test/parallel/test-worker-prof.js

```diff
@@ -5,36 +5,68 @@ const fs = require('fs');
 const assert = require('assert');
 const { join } = require('path');
 const { spawnSync } = require('child_process');
-const { Worker } = require('worker_threads');
 
 // Test that --prof also tracks Worker threads.
 // Refs: https://github.com/nodejs/node/issues/24016
 
 if (process.argv[2] === 'child') {
-  const spin = `
-  const start = Date.now();
-  while (Date.now() - start < 1000);
+  let files = fs.readdirSync(tmpdir.path);
+  const plog = files.filter((name) => /\.log$/.test(name))[0];
+  if (plog === undefined) {
+    console.error('`--prof` did not produce a profile log for parent thread!');
+    process.exit(1);
+  }
+  const pingpong = `
+  let counter = 0;
+  const { Worker, parentPort  } = require('worker_threads');
+  parentPort.on('message', (m) => {
+    if (counter++ === 1024)
+      process.exit(0);
+     parentPort.postMessage(
+       m.toString().split('').reverse().toString().replace(/,/g, ''));
+  });
   `;
-  new Worker(spin, { eval: true });
-  eval(spin);
-  return;
-}
 
-tmpdir.refresh();
-spawnSync(process.execPath, ['--prof', __filename, 'child'],
-          { cwd: tmpdir.path });
-const files = fs.readdirSync(tmpdir.path);
-const logfiles = files.filter((name) => /\.log$/.test(name));
-assert.strictEqual(logfiles.length, 2);  // Parent thread + child thread.
+  const { Worker } = require('worker_threads');
+  const data = 'x'.repeat(1024);
+  const w = new Worker(pingpong, { eval: true });
+  w.on('message', (m) => {
+    w.postMessage(m.toString().split('').reverse().toString().replace(/,/g, ''));
+  });
+
+  w.on('exit', () => {
+    files = fs.readdirSync(tmpdir.path);
+    const wlog = files.filter((name) => /\.log$/.test(name) && name !== plog)[0];
+    if (wlog === undefined) {
+      console.error('`--prof` did not produce a profile log' +
+                    ' for worker thread!');
+      process.exit(1);
+    }
+    process.exit(0);
+  });
+  w.postMessage(data);
+} else {
+  tmpdir.refresh();
```

Comment:
An absolute micronit here but maybe add a comment above this (right below the `else` line) saying this is the parent process?